networkdb/cluster.go

```diff
@@ -33,6 +34,38 @@ func (l *logWriter) Write(p []byte) (int, error) {
 	return len(p), nil
 }
 
+func (nDB *NetworkDB) SetKey(key []byte) {
+	for _, dbKey := range nDB.config.Keys {
+		if bytes.Equal(key, dbKey) {
+			return
+		}
+	}
+	nDB.config.Keys = append(nDB.config.Keys, key)
+	nDB.keyring.AddKey(key)
+}
+
+func (nDB *NetworkDB) SetPrimaryKey(key []byte) {
+	for _, dbKey := range nDB.config.Keys {
+		if bytes.Equal(key, dbKey) {
+			if nDB.memberlist != nil {
+				nDB.keyring.UseKey(dbKey)
+			}
+		}
```

Comment:
Yes, its ok to stop iterating after finding the added or removed key.

I did the same in agent.go/agentHandleKeys(). It also removes the slice entry deletion while iterating over it. PTAL.